### PR TITLE
Make H2Star work incrementally on existing data too.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: merkle/hstar2.go [hunk 1]
```diff
@@ -32,12 +32,45 @@ func NewHStar2(treeHasher TreeHasher) HStar2 {
...
+// SparseGetNodeFunc should return any pre-existing node hash for the node address.
+type SparseGetNodeFunc func(depth int, index *big.Int) (trillian.Hash, error)
+
+// SparseSetNodeFunc should store the passed node hash, associating it with the address.
+type SparseSetNodeFunc func(depth int, index *big.Int, hash trillian.Hash) error
+
+// HStar2Nodes calculates the root hash of a pre-existing sparse merkle tree plus the extra values passed in.
+// It uses the get and set functions to fetch and store updated internal node values.
+// The set parameter may be nil if no storage of updated nodes is required.
+func (s *HStar2) HStar2Nodes(n int, values []HStar2LeafHash, get SparseGetNodeFunc, set SparseSetNodeFunc) (trillian.Hash, error) {
```
Comment (Martin2112) [inline merkle/hstar2.go]: Would it be cleaner to pass an empty function and remove the if

File: merkle/hstar2_test.go [hunk 2]
```diff
@@ -23,25 +24,27 @@ func mustDecode(b64 string) trillian.Hash {
...
 }
 
-// createHStar2Values builds a list of HStar2Value structs suitable for
+// createHStar2Leaves builds a list of HStar2LeafHash structs suitable for
 // passing into a the HStar2 sparse merkle tree implementation.
 // The map keys will be SHA256 hashed before being added to the returned
 // structs.
-func createHStar2Values(vs map[string]string) []HStar2Value {
-	r := []HStar2Value{}
+func createHStar2Leaves(th TreeHasher, vs map[string]string) []HStar2LeafHash {
```
Comment (Martin2112) [inline merkle/hstar2_test.go]: 'vs'? Not obvious what this parameter is.

File: merkle/hstar2.go (hunk 1 above)
Comment (AlCutter) [inline merkle/hstar2.go]: Could be - done.

File: merkle/hstar2_test.go (hunk 2 above)
Comment (AlCutter) [inline merkle/hstar2_test.go]: "values", updated to make it clear(er)

**Pull request merged**